caffe2/utils/proto_convert.cc

```diff
@@ -0,0 +1,190 @@
+    CAFFE_ENFORCE(attr.has_f());
+    arg->set_f(attr.f());
+  } else if (
+      type ==
+      ::torch::AttributeProto_AttributeType::AttributeProto_AttributeType_INT) {
+    CAFFE_ENFORCE(attr.has_i());
+    arg->set_i(attr.i());
+  } else if (
+      type ==
+      ::torch::AttributeProto_AttributeType::
+          AttributeProto_AttributeType_STRING) {
+    CAFFE_ENFORCE(attr.has_s());
+    arg->set_s(attr.s());
+  } else if (
+      type ==
+      ::torch::AttributeProto_AttributeType::
+          AttributeProto_AttributeType_TENSOR) {
+    CAFFE_THROW("Caffe2's Argument does not support tensor as attribute.");
+  } else if (
+      type ==
+      ::torch::AttributeProto_AttributeType::
+          AttributeProto_AttributeType_GRAPH) {
+    // TODO
+    CAFFE_THROW("GraphProto conversion is not implemented yet.");
+  } else if (
+      type ==
+      ::torch::AttributeProto_AttributeType::
+          AttributeProto_AttributeType_FLOATS) {
+    arg->mutable_floats()->CopyFrom(attr.floats());
+  } else if (
+      type ==
+      ::torch::AttributeProto_AttributeType::
+          AttributeProto_AttributeType_INTS) {
+    arg->mutable_ints()->CopyFrom(attr.ints());
+  } else if (
+      type ==
+      ::torch::AttributeProto_AttributeType::
+          AttributeProto_AttributeType_STRINGS) {
+    arg->mutable_strings()->CopyFrom(attr.strings());
+  } else if (
+      type ==
+      ::torch::AttributeProto_AttributeType::
+          AttributeProto_AttributeType_TENSORS) {
+    CAFFE_THROW("Caffe2's Argument does not support tensors as attribute.");
+  } else if (
+      type ==
+      ::torch::AttributeProto_AttributeType::
+          AttributeProto_AttributeType_GRAPHS) {
+    // TODO
+    CAFFE_THROW("GraphProtos conversion is not implemented yet.");
+  } else {
+    CAFFE_THROW("Unknow Attribute type.");
+  }
+}
+
+CAFFE2_EXPORT void OperatorDefToNodeProto(
+    const OperatorDef& def,
+    ::torch::NodeProto* node) {
+  node->mutable_input()->CopyFrom(def.input());
+  node->mutable_output()->CopyFrom(def.output());
+  if (def.has_name()) {
+    node->set_name(def.name());
+  }
+  CAFFE_ENFORCE(def.has_type());
+  node->set_op_type(def.type());
+  for (int i = 0; i < def.arg_size(); ++i) {
+    auto attr = node->add_attribute();
+    ArgumentToAttributeProto(def.arg(i), attr);
+  }
+  if (def.has_device_option()) {
+    node->mutable_device_option()->CopyFrom(def.device_option());
+  }
+  if (def.has_engine()) {
+    auto attr = node->add_annotations();
+    attr->set_name("engine");
+    attr->set_type(::torch::AttributeProto_AttributeType::
+                       AttributeProto_AttributeType_STRING);
+    attr->set_s(def.engine());
+  }
+  if (def.control_input_size() > 0) {
+    auto attr = node->add_annotations();
+    attr->set_name("control_input");
+    attr->set_type(::torch::AttributeProto_AttributeType::
+                       AttributeProto_AttributeType_STRINGS);
+    for (int i = 0; i < def.control_input_size(); ++i) {
```

Comment:
attr->mutable_strings()->CopyFrom(def.control_input());
or
for (const string& input : def.control_input()) {
  attr->add_strings(input);
}